lms/djangoapps/support/views/course_entitlements.py

```diff
@@ -0,0 +1,144 @@
+"""
+Support tool for changing and granting course entitlements
+"""
+from django.contrib.auth.models import User
+from django.db.models import Q
+from django.http import HttpResponseBadRequest
+from django.utils.decorators import method_decorator
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions, status, viewsets
+from rest_framework.response import Response
+from six import text_type
+
+from entitlements.api.v1.permissions import IsAdminOrAuthenticatedReadOnly
+from entitlements.api.v1.serializers import SupportCourseEntitlementSerializer
+from entitlements.models import CourseEntitlement, CourseEntitlementSupportDetail
+from openedx.core.djangoapps.cors_csrf.authentication import SessionAuthenticationCrossDomainCsrf
+from lms.djangoapps.support.decorators import require_support_permission
+
+
+class EntitlementSupportView(viewsets.ModelViewSet):
+    """
+    Allows viewing and changing learner course entitlements, used the support team.
+    """
+    authentication_classes = (JwtAuthentication, SessionAuthenticationCrossDomainCsrf,)
+    permission_classes = (permissions.IsAuthenticated, IsAdminOrAuthenticatedReadOnly,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = SupportCourseEntitlementSerializer
+
+    def filter_queryset(self, queryset):
+        try:
+            username_or_email = self.request.GET.get('user')
+            user = User.objects.get(Q(username=username_or_email) | Q(email=username_or_email))
+            queryset = queryset.filter(user=user)
+            return super(EntitlementSupportView, self).filter_queryset(queryset).order_by('created')
+        except (KeyError, User.DoesNotExist):
+            return queryset.order_by('created')
+
+    @method_decorator(require_support_permission)
+    def update(self, request):
+        """ Allows support staff to unexpire a user's entitlement."""
+        support_user = request.user
+        try:
+            username_or_email = self.request.data['user']
```

Comment:
Two concerns here

The first is the use of user for both username and email.  The second is that I would use a get for the request data. If there is not user in the data you will get an exception with the current access method. We should avoid using Exceptions as error handling when we can.

Reply:
Is it not reasonable to catch the key error and indicate to the end user that the field not included in the request is required?